src/map.cpp

```diff
@@ -13,11 +13,15 @@
 
 namespace bpftrace {
 
-int Map::create_map(enum bpf_map_type map_type, const char *name, int key_size, int value_size, int max_entries, int flags) {
+int Map::create_map(enum bpf_map_type map_type, const std::string & name, int key_size, int value_size, int max_entries, int flags) {
+  std::string name_ = name;
+  if (name_[0] == '@') {
+    name_ = "AT_" + name.substr(1);
+  }
 #ifdef HAVE_BCC_CREATE_MAP
-  return bcc_create_map(map_type, name, key_size, value_size, max_entries, flags);
+  return bcc_create_map(map_type, name_.c_str(), key_size, value_size, max_entries, flags);
 #else
-  return bpf_create_map(map_type, name, key_size, value_size, max_entries, flags);
+  return bpf_create_map_name(map_type, name_.c_str(), key_size, value_size, max_entries, flags);
```

Comment:
The `#ifdef HAVE_BCC_CREATE_MAP` is to handle when bcc renamed its own `bpf_create_map` -> `bcc_create_map`. Changing this to libbpf's `bpf_create_map_name` may have unintended consequences when bpftrace is compiled without libbpf.

All that to say: please undo this change :)

Reply:
Done.